core/common/src/main/java/alluxio/resource/RefCountLockResource.java

```diff
@@ -63,9 +52,6 @@ public RefCountLockResource(ReentrantReadWriteLock lock, LockMode mode, boolean
   @Override
   public void close() {
     super.close();
```

Comment:
`RefCountLockResource` direct parent is `RWLockResource` which has no `close()`, so I think this calls the un-overriden `close()` from `LockResource`. 

If this gives you a sense of insecurity, we can add a dummy `close()` to `RWLockResource` to make sure
```
@Override
public void close() {super.close();}
```

Reply:
I manually set the lock pool high watermark to very low to trigger lock eviction. But I don't see any complaint on lock leaks. So I think the current approach is fine.

To stay on the safe side, I think adding a dummy `close()` to `RWLockResource` feels the most secure.

Reply:
No need for dummy methods if the behavior is what you defined here.

Reply:
BTW, I don't think lowering the pool size will trigger the leak. The leak was happening when OOM errors was seen. After @ZhuTopher's fixes, I've never seen a report of such leaks. Could still be happening though...

Reply:
Verified the behavior is as we expected, the propagation works:
1. RefCountLockResource.close() calls super.close()
1. RWLockResource does not override close(), so LockResource.close() is actually called. This is what we want.

No need for the dummy method, I will remove that.